frontend/apps/reader/modules/readersearch.lua

```diff
@@ -496,7 +496,8 @@ function ReaderSearch:showFindAllResults()
         onMenuSelect = function(self_menu, item)
             self_menu.close_callback()
             if self.ui.rolling then
-                self.ui.link:onGotoLink({ xpointer = item.start })
+                self.ui.link:addCurrentLocationToStack()
+                self.ui.rolling:onGotoXPointer(item.start, item.start) -- show target line marker
                 self.ui.document:getTextFromXPointers(item.start, item["end"], true) -- highlight
```

Comment:
Do we need to show both the line marker, and the text highlighted ? (If I get right what you want with this change).

Reply:
If there are highlights in the target page, it's hard to notice the highlighted search result.